setup.py

```diff
@@ -1,11 +1,23 @@
+import sys
+import os
+import json
+
+#Required version of python interpreter
+req_version = (3, 5)
+#Current version of python interpreter
+curr_version = sys.version_info
+
+#Exit if minimum requirements are not met
+if curr_version < req_version:
+    raise SystemExit("Your python interpreter does not meet" +
+                     " the minimum requirements.\n" +
+                     "Consider upgrading to python3.5")
+
 from setuptools import setup
```

Comment:
Ok, but this one seemed better, since we don't need `setuptools` installed before determining the correct version... also, setuptools need to be installed manually and can't be included in the requirements.txt, so, I think this is the way we should proceed.